### PR TITLE
add hybrid dev docs

### PR DESCRIPTION
**Pull request by AdrianPedriza**
documentation about the new feature released on hybrid mode development.

### Proposed changes

- add new page under `okteto manifest` section with information about what is hybrid development
- add docs about new fields in manifest

### CODE REVIEW

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [inline src/content/manifest/hybrid-development.mdx:22]: @rlamana should we include an image about the arch here? similar to https://github.com/okteto/okteto/blob/master/docs/okteto-architecture.png

Comment (rlamana) [reply src/content/manifest/hybrid-development.mdx:22]: If I create a diagram, I think we should also add a description of the how, to me this paragraph talks more about what it is rather that how it is achieved...

Comment (rlamana) [reply src/content/manifest/hybrid-development.mdx:22]: I found a diagram we already had something like this:

Screenshot 2023-12-14 at 15 39 12

Comment (pchico83) [reply src/content/manifest/hybrid-development.mdx:22]: @rlamana could we also have the image representing the state before `okteto up`? and in this image, I would replace the `api` text inside the namespace by `SSH Proxy`

Comment (rlamana) [reply src/content/manifest/hybrid-development.mdx:22]: @pchico83 What about something like this:

Screenshot 2023-12-15 at 12 01 07

File: src/content/manifest/hybrid-development.mdx
Comment (rlamana) [inline src/content/manifest/hybrid-development.mdx:23]: Not sure if this paragraph is clear to me. Let's try to define it in a way that explains what is shown in the diagram.

Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: ok! take a look at this paragraph and let me know if it is ok https://github.com/okteto/docs/pull/557/files#diff-4704534dddb0360cc8e8c8559f984492563b94235123bf13714fd4867ad9933aR23

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx:22]: I've added the last one ATM

**Comment by huguestennier** [comment]
Before this goes live, please moves all the images into the `static/img` folder and note how we need to reference images now:

```
 <Image src={require("@site/static/img/compose-endpoints.png").default} />
  ```
  
  More details in https://github.com/okteto/docs/pull/558

File: src/content/manifest/hybrid-development.mdx
Comment (huguestennier) [inline src/content/manifest/hybrid-development.mdx:10]:
```suggestion
Okteto offers two different approaches to service development: *sync* ot *hybrid* mode. The command affected by this is the [`okteto up`](reference/cli.mdx#up) command being configurable at manifest level.
```

You need to link directly to the .mdx file, else this will break when we do versioning.

More details here: https://github.com/okteto/docs/pull/295

Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: great 041bf0170a82d66651c665dd74636951a932d0bc

File: src/content/manifest/hybrid-development.mdx
Comment (omnomagonz) [inline src/content/manifest/hybrid-development.mdx]:
```suggestion
Okteto offers two different approaches to service development: `sync` or `hybrid` mode. This mode is configured in your Okteto manifest and affects the behavior of [`okteto up`](/reference/cli/#up). 
```

File: src/content/manifest/hybrid-development.mdx
Comment (omnomagonz) [inline src/content/manifest/hybrid-development.mdx:12]:
```suggestion
`Sync` is the default mode and changes made to the local service are synchronized with the remote development container deployed in the cluster. Alternatively, `hybrid` mode enables you to run the service in development locally while keeping the rest of the application components in the cloud.
```

File: src/content/manifest/hybrid-development.mdx
Comment (omnomagonz) [inline src/content/manifest/hybrid-development.mdx]:
```suggestion
By default Okteto enables you to edit a service's code locally and have it run in a production-like cloud environment. This is achieved using `syncthing` where local changes are synchronized with the development container running in the cloud.

Okteto also allows you to use another mode called `hybrid` in which the service to be developed runs locally and the rest of the application components run in the cloud.
```

I'm wondering if we could just omit the first paragraph especially if it's mentioned elsewhere. It seems a bit odd to have a section header called "How does hybrid mode work" and then spent the first half talking about sync mode 🤔 

any opinions?

File: src/content/manifest/hybrid-development.mdx
Comment (omnomagonz) [inline src/content/manifest/hybrid-development.mdx:14]:
```suggestion
### Deciding which mode to use
```

Maybe something like this? "Problem to solve" doesn't seem fitting for public docs since "problem" has a negative connotation. I propose something like this instead 😄

File: src/content/manifest/hybrid-development.mdx
Comment (omnomagonz) [inline src/content/manifest/hybrid-development.mdx:16]:
```suggestion
There are situations when running a service locally is faster than synchronizing code with a remote development container. For example: when doing frontend-only development or when working in an IDE-heavy context (e.g. Java/Spring) where the IDE may be pre-configured to run tests, start services, and connect debuggers. 

When using Okteto's `hybrid` mode you can maintain your local development experience for the specific service you're working on while still leveraging the benefits of hosting the rest of your application's components in the remote development container.
```

Wondering if something like this might work?

File: src/content/manifest/hybrid-development.mdx
Comment (omnomagonz) [inline src/content/manifest/hybrid-development.mdx:20]:
```suggestion
### How hybrid development works
```

File: src/content/manifest/hybrid-development.mdx
Comment (omnomagonz) [inline src/content/manifest/hybrid-development.mdx:21]:
```suggestion
To allow communication to and from the cluster and your local machine an SSH server is deployed in the remote development container. You can expose the local service to the development container using the [`reverse`](/reference/okteto-manifest.mdx/#reverse-string-optional) configuration for the development environment. If you want to forward a service exposed in the cluster to your local machine you can use the [`forward`](/reference/okteto-manifest.mdx/#forward-string-optional) configuration.
```

File: src/content/manifest/hybrid-development.mdx
Comment (omnomagonz) [inline src/content/manifest/hybrid-development.mdx:31]:
```suggestion
When using `hybrid` mode the process running locally is designed to behave similarly as if it were running in `sync` mode. To achieve this the following environment variables are added to the local process running the service in `hybrid` mode. This list is ordered from highest to lowest prevalence:
```

File: src/content/manifest/hybrid-development.mdx
Comment (omnomagonz) [inline src/content/manifest/hybrid-development.mdx:32]:
```suggestion
- Environment variables declared in the `dev` section of the manifest
```

File: src/content/manifest/hybrid-development.mdx
Comment (omnomagonz) [inline src/content/manifest/hybrid-development.mdx:35]:
```suggestion
- User-defined variables
```

@pchico83 suggesting this change as part of making "secrets" -> "variables" language more consistent

File: src/content/manifest/hybrid-development.mdx
Comment (omnomagonz) [inline src/content/manifest/hybrid-development.mdx:38]:
```suggestion
### How to develop in hybrid mode
```

File: src/content/manifest/hybrid-development.mdx
Comment (omnomagonz) [inline src/content/manifest/hybrid-development.mdx:40]:
```suggestion
To enable `hybrid` mode you only need to configure the necessary ports to expose the local service to the remote development container. This allows the components running in the cluster to communicate with the local service as if it was running directly in the cluster. 

To do this you can make use of the [reverse](reference/manifest/#reverse-string-optional) field in your Okteto manifest, which will expose the configured port of your local machine to the desired port on the development container.
```

File: src/content/manifest/hybrid-development.mdx
Comment (omnomagonz) [inline src/content/manifest/hybrid-development.mdx:52]:
```suggestion
In this example we have configured the `frontend` service to use `mode: hybrid`. This means the `frontend` service will exist on our local machine and not in the development container. 

This service will execute (locally) the command defined in the `command` field and in the directory specified in `workdir`. Additionally, anything exposed locally on port `8080` will be exposed on port `3000` of the development container. This makes it possible for other components running on the cluster to access the local service as if it were running remotely in the cluster.
```

@pchico83 mind reviewing this to make sure I haven't misunderstood something? 🙏

File: src/content/reference/okteto-manifest.mdx
Comment (omnomagonz) [inline src/content/reference/okteto-manifest.mdx:533]:
```suggestion
- `sync`: the standard and default mode. This mode will synchronize your code with the remote development container
```

File: src/content/reference/okteto-manifest.mdx
Comment (omnomagonz) [inline src/content/reference/okteto-manifest.mdx:534]:
```suggestion
- `hybrid`: use this mode if you want to run your service locally but run the rest of your application components in the cluster
```

File: src/content/manifest/hybrid-development.mdx
Comment (pchico83) [reply src/content/manifest/hybrid-development.mdx:52]: The second paragraph looks great, but in the first one I wouldn't say "`frontend` service will exist on our local machine", I would say something "okteto will redirect requests to the `frontend` service to your local machine.
And the second paragraph could say:
"Okteto will execute (locally) the command..."

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: 1d810a59e441c6d926b8b04c03046b45b82df0d2

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: agree

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: 3b791f1eb3ff23bf057a17da1065f5533b18133a

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: af9a32b317861e294945db0b4e0dd006fdc2c9e8

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: 1bbd05c534a2e3b502ffb109e5c180001b530e66

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: d43a8d7150223e44aa96a02b8d7c57b5e45deb13

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: ce55671f47de5b99d0755d85353d8ac2a21101ff

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: a261c50d657ca80b711e92a88c878e5865bc1943

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: 3eba66b255a3178d3016b3dda7a8576cc10a860c

File: src/content/reference/okteto-manifest.mdx
Comment (AdrianPedriza) [reply src/content/reference/okteto-manifest.mdx]: 61c68a673f570641b94f33829c8bacb7a7c4d210

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: 768595236ccf3f4be66c9cb95b547cd7d0de78b2

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: check the preview after the changes in that section. Are you OK with it?

**Review by omnomagonz — APPROVED** [review]

File: src/content/manifest/hybrid-development.mdx
Comment (rlamana) [reply src/content/manifest/hybrid-development.mdx:21]: @pchico83 can you review the diagram?

**Review by rlamana — APPROVED** [review]
LGTM but I would wait for approval on the diagram

File: src/content/manifest/hybrid-development.mdx
Comment (pchico83) [inline src/content/manifest/hybrid-development.mdx:10]: @omnomagonz not in this PR, but I think this page should explain how dev containers work. And as part of it, explain how `sync` and `hybrid` modes work

File: src/content/manifest/hybrid-development.mdx
Comment (pchico83) [inline src/content/manifest/hybrid-development.mdx:20]:
```suggestion
> Using `hybrid` might increase latency when accessing the rest of the services in the cluster. 
```
I would remove the mention to applicability. This can be solved by running `okteto up` inside a local container

File: src/content/manifest/hybrid-development.mdx
Comment (pchico83) [inline src/content/manifest/hybrid-development.mdx:24]:
```suggestion
To allow communication to and from the cluster and your local machine an SSH server is deployed in the remote development container. You can expose the local service to the services running in the cluster using the [`reverse`](/reference/okteto-manifest.mdx/#reverse-string-optional) configuration for the development environment. If you want to forward a service running in the cluster to your local machine you can use the [`forward`](/reference/okteto-manifest.mdx/#forward-string-optional) configuration.
```

File: src/content/manifest/hybrid-development.mdx
Comment (pchico83) [inline src/content/manifest/hybrid-development.mdx:34]:
```suggestion
When using `hybrid` mode the process running locally is designed to run with the same environment as the original service running in the cluster. To achieve this, the following environment variables are added to the local process running the service in `hybrid` mode. This list is ordered from highest to lowest prevalence:
```

File: src/content/manifest/hybrid-development.mdx
Comment (pchico83) [inline src/content/manifest/hybrid-development.mdx:36]:
```suggestion
- Environment variables defined in the original container spec (add link to container envars k8s docs)
```

File: src/content/manifest/hybrid-development.mdx
Comment (pchico83) [inline src/content/manifest/hybrid-development.mdx:37]:
```suggestion
- Environment variables coming from configmaps defined in the original container spec (add link to k8s docs)```

File: src/content/manifest/hybrid-development.mdx
Comment (pchico83) [inline src/content/manifest/hybrid-development.mdx:38]:
```suggestion
- Environment variables coming from secrets defined in the original container spec (add link to k8s docs)
```

File: src/content/manifest/hybrid-development.mdx
Comment (pchico83) [inline src/content/manifest/hybrid-development.mdx:39]:
```suggestion
- Environment variables declared in the image of the original container (k8s docs link to the field `image`)
```

File: src/content/manifest/hybrid-development.mdx
Comment (pchico83) [inline src/content/manifest/hybrid-development.mdx:43]:
```suggestion
To enable `hybrid` mode you only need to configure the necessary ports to expose the local service to the cluster network. This allows the components running in the cluster to communicate with the local service as if it was running directly in the cluster. 
```

File: src/content/manifest/hybrid-development.mdx
Comment (pchico83) [inline src/content/manifest/hybrid-development.mdx:57]:
```suggestion
In this example we have configured the `frontend` service to use `mode: hybrid`. This means Okteto will redirect requests to the frontend service in the cluster to your local machine.
```

File: src/content/manifest/hybrid-development.mdx
Comment (pchico83) [inline src/content/manifest/hybrid-development.mdx:59]: `Okteto will execute (locally) the command defined in the `command` field and in the directory specified in `workdir` -> I would mention here that we also autoinject environment variables to simulate the same environment than the original container. With a link to where this is documented

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: 9ed9c2367e2fee70d1c8a8e1a553175cea6459fd

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: 359692e888a6f51dc3f0bd5026832fbd7c780f10

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx:57]: 8508a5d2898cebc9c26ac3b52202ae30c5d1cd54

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: 8508a5d2898cebc9c26ac3b52202ae30c5d1cd54

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: a52b4aea3ea4ea89f11c21d64b43d6526b200670

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx:39]: a link to https://kubernetes.io/docs/concepts/containers/images/ is enough?

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx:38]: what link should we use here?

Comment (pchico83) [reply src/content/manifest/hybrid-development.mdx:38]: https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#define-container-environment-variables-using-secret-data

Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: c254fd0cc7e0892e68bede41146296585d0f0e8e

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: c254fd0cc7e0892e68bede41146296585d0f0e8e

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: c254fd0cc7e0892e68bede41146296585d0f0e8e

File: src/content/manifest/hybrid-development.mdx
Comment (pchico83) [inline src/content/manifest/hybrid-development.mdx:12]:
```suggestion
`Sync` is the default mode and code changes made to the local filesystem are synchronized to the remote development container deployed in the cluster. Alternatively, `hybrid` mode enables you to run the service in development locally while keeping the rest of the application components in the cloud.
```

File: src/content/manifest/hybrid-development.mdx
Comment (pchico83) [inline src/content/manifest/hybrid-development.mdx:35]: @AdrianPedriza can we add a link to https://www.okteto.com/docs/reference/manifest/#environment-string-optional?

File: src/content/manifest/hybrid-development.mdx
Comment (pchico83) [reply src/content/manifest/hybrid-development.mdx:37]: @AdrianPedriza I think you missed this comment

**Review by pchico83 — APPROVED** [review]
@AdrianPedriza 2 minor comments, but ready to merge :-)

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: ae356f614884e5efb1be8236b4369c11f6af7f6b

File: src/content/manifest/hybrid-development.mdx
Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx:37]: oh yes I overlooked it. What k8s docs would be appropriate to link here?

Comment (pchico83) [reply src/content/manifest/hybrid-development.mdx:37]: this is a good link:
https://kubernetes.io/es/docs/concepts/configuration/configmap/#configmaps-y-pods

Comment (AdrianPedriza) [reply src/content/manifest/hybrid-development.mdx]: great 5bc16e5b1befd0fa22206247f20ecf5a480dc0a9

**Pull request merged**
